Initial update of docs to utilize Butane configs (#260)
flatcar/flatcar-docs: docs/provisioning/config-transpiler/_index.md

## Tell me more about Ignition

[Ignition][ignition] is the utility inside of a Container Linux image that is responsible for setting up a machine. It takes in a configuration, written in JSON, that instructs it to do things like add users, format disks, and install systemd units. The artifacts that butane produces are Ignition configs. All of this should be an implementation detail however, users are encouraged to write Butane Configs for butane, and to simply pass along the produced JSON file to their machines.

## How similar are Butane Configs and Ignition configs?

Some features in Butane Configs and Ignition configs are identical.  Both support listing users for creation, systemd unit dropins for installation, and files for writing.

All of the differences stem from the fact that Ignition configs are distribution agnostic. An Ignition config can't just tell Ignition to enable etcd, because Ignition doesn't know what etcd is. The config must tell Ignition what systemd unit to enable, and provide a systemd dropin to configure etcd.

> Butane on the other hand _does_ understand the specifics of Container Linux. A user can merely specify an etcd version and some etcd options, and butane knows that there's a unit called `etcd-member` already on the system it can enable. It knows what options are supported by etcd, so it can sanity check them for the user. It can then generate an appropriate systemd dropin with the user's options, and provide Ignition the level of verbosity it needs, that would be tedious for a human to create.

Reviewer: I think this not the case with `butane`. It's not aware about etcd-member, locksmith, etc.

Reviewer 2: Butane's Flatcar variant is aware of Flatcar specifics, maybe we can talk about this (currently it just means that generating clevis sections is caught because they are not supported)

Reviewer 2: ```suggestion
Butane on the other hand _does_ understand the specifics of Flatcar Container Linux. A user currently can't specify Clevis options on Flatcar and Butane does these sanity checks for the user.
```

## Example Butane Config

The following small example of a Butane Config will ensure that the default core user exists and adds a specified public key

```yaml
variant: flatcar
version: 1.0.0
passwd:
  users:
    - name: core
      ssh_authorized_keys:
        - ssh-rsa AAAAB......xyz email@host.net
```

> To turn this Butane Config into a usable Ignition Config, we can then run: `docker run --rm -i quay.io/coreos/butane:latest`. This will result in the above YAML being turned into the below JSON

Reviewer: ```suggestion
To turn this Butane Config into a usable Ignition Config, we can then run: `docker run --rm -i quay.io/coreos/butane:latest < your_config.yaml > your_config.json`. This will result in the above YAML being turned into the below JSON
```
